### PR TITLE
Add data directory excluding sessions to dump (#586)

### LINKED ISSUE #586: `Dump` should contains data directory except sessions
Currently `Dump` command didn't contain data directory. It will missing the important content when restore in future.

### PR DESCRIPTION
**Pull request by mkuhlmann**
This includes the data directory inside a gitea dump. Sessions are excluded if the session provider is set to file and the session directory is a child of the data directory.

Closes #586

### CODE REVIEW

**Comment by mkuhlmann** [comment]
Works now. The issue was an unset `PROVIDER_CONFIG` in custom app.ini. Then the configuration value is just empty and macaron uses a standard `data/sessions`. Thats another bug that should be resolved, since the expected behavior should session path to be relative to `APP_DATA_PATH` and not to the executable path. If `APP_DATA_PATH` is set to another location but `PROVIDER_CONFIG` is not set, sessions will be still saved to `data/sessions`.

I've hardcoded `data/sessions` to match [macarons](https://github.com/go-gitea/gitea/blob/9d1bc9aac88bda13e4b9319ab915bca1b65521f6/vendor/github.com/go-macaron/session/session.go#L113) fallback config var as a workaround and it works for now.

**Comment by lunny** [comment]
LGTM

File: cmd/dump.go [hunk 1]
```diff
@@ -129,3 +145,44 @@ func runDump(ctx *cli.Context) error {
...
+	dir, err := os.Open(absPath)
+	if err != nil {
+		return err
+	}
+	defer dir.Close()
+
+	// Make sure we have all upper level directories.
+	z.AddEmptyDir(dirPath)
+
+	fis, err := dir.Readdir(0)
```
Comment (bkcsoft) [inline cmd/dump.go]: could we call it `files, err` instead?

File: cmd/dump.go [hunk 2]
```diff
@@ -129,3 +145,44 @@ func runDump(ctx *cli.Context) error {
...
+	if err != nil {
+		return err
+	}
+	for _, fi := range fis {
```
Comment (bkcsoft) [inline cmd/dump.go]: same hear, `file := range files`

**Review by bkcsoft — CHANGES_REQUESTED** [review]
Please don't use uncommon abbreviations for variables :)

**Comment by mkuhlmann** [comment]
@bkcsoft Fixed, much cleaner now ;)

**Comment by strk** [comment]
LGTM

**Comment by lunny** [comment]
We have to merge this at first. @bkcsoft if you have any further information, please open another issue.

**Pull request merged**